Do not create and resolve directories for resolved directories in WebsiteDataStore

szewai opened the pull request:
#### dc2259320c914a459734ec48128cdd4a1ed6a43e
<pre>
Do not create and resolve directories for resolved directories in WebsiteDataStore
<a href="https://bugs.webkit.org/show_bug.cgi?id=242803">https://bugs.webkit.org/show_bug.cgi?id=242803</a>
rdar://96917273

Reviewed by Geoffrey Garen.

We cache resolved directories in WebsiteDataStore, so we can use them to create sandbox extension handle without
resolving them again. This should help reduce some syscalls.

* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::webProcessDataStoreParameters):
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::cacheDirectoryInContainerOrHomeDirectory):
(WebKit::WebsiteDataStore::parentBundleDirectory const):
(WebKit::WebsiteDataStore::resolvedCookieStorageDirectory):
(WebKit::WebsiteDataStore::resolvedContainerCachesNetworkingDirectory):
(WebKit::WebsiteDataStore::resolvedContainerCachesWebContentDirectory):
(WebKit::WebsiteDataStore::resolvedContainerTemporaryDirectory):
(WebKit::WebsiteDataStore::defaultContainerTemporaryDirectory):
(WebKit::WebsiteDataStore::cookieStorageDirectory const): Deleted.
(WebKit::WebsiteDataStore::containerCachesDirectory const): Deleted.
(WebKit::WebsiteDataStore::networkingCachesDirectory const): Deleted.
(WebKit::WebsiteDataStore::containerTemporaryDirectory const): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):
(WebKit::createHandleFromResolvedPathIfPossible):
(WebKit::WebsiteDataStore::parameters):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
(WebKit::WebsiteDataStore::resolvedCacheStorageDirectory const):
(WebKit::WebsiteDataStore::resolvedPrivateClickMeasurementStorageDirectory const):
(WebKit::WebsiteDataStore::cacheStorageDirectory const): Deleted.

Canonical link: <a href="https://commits.webkit.org/252534@main">https://commits.webkit.org/252534@main</a>
</pre>

geoffreygaren reviewed (approved):
r=me

Note that there's also a slight behavior change: If you modify a symlink while a WebsiteDataStore is in use, the WebsiteDataStore will ignore the modification and continue using the path it resolved before. 

Hard to say exactly what correct behavior would be in that situation. Continuing to use the same path seems like a slight improvement to me, since the alternative would likely produce data "tearing".

szewai committed:
Do not create and resolve directories for resolved directories in WebsiteDataStore
https://bugs.webkit.org/show_bug.cgi?id=242803
rdar://96917273

Reviewed by Geoffrey Garen.

We cache resolved directories in WebsiteDataStore, so we can use them to create sandbox extension handle without
resolving them again. This should help reduce some syscalls.

* Source/WebKit/UIProcess/WebProcessPool.cpp:
(WebKit::WebProcessPool::webProcessDataStoreParameters):
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::cacheDirectoryInContainerOrHomeDirectory):
(WebKit::WebsiteDataStore::parentBundleDirectory const):
(WebKit::WebsiteDataStore::resolvedCookieStorageDirectory):
(WebKit::WebsiteDataStore::resolvedContainerCachesNetworkingDirectory):
(WebKit::WebsiteDataStore::resolvedContainerCachesWebContentDirectory):
(WebKit::WebsiteDataStore::resolvedContainerTemporaryDirectory):
(WebKit::WebsiteDataStore::defaultContainerTemporaryDirectory):
(WebKit::WebsiteDataStore::cookieStorageDirectory const): Deleted.
(WebKit::WebsiteDataStore::containerCachesDirectory const): Deleted.
(WebKit::WebsiteDataStore::networkingCachesDirectory const): Deleted.
(WebKit::WebsiteDataStore::containerTemporaryDirectory const): Deleted.
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):
(WebKit::createHandleFromResolvedPathIfPossible):
(WebKit::WebsiteDataStore::parameters):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
(WebKit::WebsiteDataStore::resolvedCacheStorageDirectory const):
(WebKit::WebsiteDataStore::resolvedPrivateClickMeasurementStorageDirectory const):
(WebKit::WebsiteDataStore::cacheStorageDirectory const): Deleted.

Canonical link: https://commits.webkit.org/252534@main
